### PR TITLE
Add codespell action

### PR DESCRIPTION
**Pull request by npentrel**
Adding a spell checker, has found some issues and I didn't have to add a great number of ignores which seems good

### CODE REVIEW

File: docs/appendix/try-viam/rover-resources/rover-tutorial/_index.md [hunk 1]
```diff
@@ -292,7 +292,7 @@ You can adjust this value if using a battery that has an alternative nominal vol
 To set the low voltage cutoff and reconnect voltage:
 
 1.  Turn on the circuit using the switch.
-    The LED indicater should indicate a current voltage level of 14.8-16V depending on the battery charge status:
+    The LED indicator should indicate a current voltage level of 14.8-16V depending on the battery charge status:
```
Comment (npentrel) [inline docs/appendix/try-viam/rover-resources/rover-tutorial/_index.md:295]:
```suggestion
    The LED indicater should indicate a current voltage level of 14.8-16V depending on the battery charge status:
```

File: docs/services/ml/upload-training-script.md [hunk 2]
```diff
@@ -37,7 +37,7 @@ Parse these arguments in your training script with the following method:
 ```python {class="line-numbers linkable-line-numbers" data-line="364"}
 # This parses the required args for the training script.
 # The model_dir variable will contain the output directory where
-# the ML model that this scrips creates should be stored.
+# the ML model that this scripts creates should be stored.
```
Comment (sguequierre) [inline docs/services/ml/upload-training-script.md:40]:
```suggestion
# the ML model that this script creates should be stored.
```

File: docs/use-cases/create-module.md [hunk 3]
```diff
@@ -514,7 +514,7 @@ var (
 )
 
 const (
-    myBaseWidthMm        = 500.0 // Base has a wheel tread of 500 millimeters
+    myBaseWidthMm        = 500.0 // Base has a wheel thread of 500 millimeters
```
Comment (sguequierre) [inline docs/use-cases/create-module.md:517]:
```suggestion
    myBaseWidthMm        = 500.0 // Base has a wheel tread of 500 millimeters
```
I think this is still correct in context

**Review by sguequierre — APPROVED** [review]
Awesome, thanks so much Naomi! This'll be great. Flagged a couple minor issues.

File: .codespellignore [hunk 4]
```diff
@@ -0,0 +1,3 @@
+SME
+AbsoluteY
+DOUT
```
Comment (npentrel) [inline .codespellignore:3]:
```suggestion
DOUT
tread
```

**Pull request merged**